brainstorming.md

```diff
@@ -0,0 +1,508 @@
+| `obs_ms`, `obsm`, `obsp`, `var_ms`, `varm`, `varp` | Fields may be empty collections.                                                                                                                                                                                                     |
+| `X` collection values                              | All matrixes must have the shape `(#obs, #var)`. The domain of the first dimension is the values of `obs.__rowid`, and the index domain of the second dimension is the values of `var.__rowid` in the containing `SOMAMeasurements`. |
+| `obsm` collection values                           | All matrixes must have the shape `(#obs, M)`, where `M` is user-defined. The domain of the first dimension is the values of `obs.__rowid`.                                                                                           |
+| `obsp` collection values                           | All matrixes must have the shape `(#obs, #obs)`. The domain of both dimensions is the values of `obs.__rowid`.                                                                                                                       |
+| `obs_ms` collection values                         | All dataframes must have the shape `(#obs,)`. The domain of the dimension is the values of `obs.__rowid`.                                                                                                                            |
+| `varm` collection values                           | All matrixes must have the shape `(#var, M)`, where `M` is user-defined. The domain of the first dimension is the values of `var.__rowid`.                                                                                           |
+| `varp` collection values                           | All matrixes must have the shape `(#var, #var)`. The domain of both dimensions is the values of `var.__rowid`.                                                                                                                       |
+| `var_ms` collection values                         | All dataframes must have the shape `(#var,)`. The domain of the dimension is the values of `var.__rowid`.                                                                                                                            |
+
+> ⚠️ **Issue** `raw` -- do we need `raw.X` and `raw.var` at all, or can we model them with existing structures? **Proposal**: the current `X` and `varm` can represent typical `raw` semantics (filtered subset, or transformed values). Assume that higher level usage conventions will utilize these to represent raw, eg, `X['raw']` and `varm['raw']`, or alternatively by adding additional columns to `var` that indicate filtering status (eg, `var['is_filtered']`).
```

Comment:
i think (disclaimer: "i am not a biologist") we need to decide where to put raw sooner than later. it's a central, immediate use-case and we just need to decide and stick.

as this PR has evolved, it's been marvelous to see our thinking evolve and crystallize into a well-thought-out way to handle multimodal data -- a pinch-point for cross-language/cross-tool operability -- to the point that we have names and a diagram which demonstrates our well-thought-out plan.

we need to achieve the same for raw.

your suggestion here is good -- I've no action item _for you_ on this -- my input is that we, collectively, need accept/counterpropose on this great suggestion of yours, sooner than later.

one concern i have (again "i am not a biologist") is what if a single experiment, with more than one assay type, has two different assays each of which has raw data? i'm not enough of a biologist to answer that question but i'm enough of an engineer to ask it ...

Reply:
Added to the issues section:
> 5. Should `raw` vs `processed` conventions be codified in this spec (eg, as predefined fields in SOMAExperiment/SOMAMeasurement)? Or should they be the purview of higher level schema/conventions, built upon the general-purpose features of this API?